stellarphot/core.py

```diff
@@ -62,3 +64,307 @@ def __init__(self, gain=1.0 * u.electron / u.adu,
         self.gain = gain
         self.read_noise = read_noise
         self.dark_current = dark_current
+
+
+class BaseEnhancedTable:
+    """
+    A base class to hold an `astropy.table.Table` data table and extend it
+    so that chosen columns can be accessed as attributes.  This base class
+    imposes the requirement that the table have columns corresponding to
+    attributes 'id', 'ra', and 'dec' attributes, as well as confirming
+    the 'ra' and 'dec' are in degrees.
+
+    Parameters
+    ----------
+
+    table_descript: `numpy.ndarray`
+        This is a 2-D numpy array where each row of the array is the
+        table column name, dtype, astropy unit (can be None), and
+        the associated attribute name (can also be None, in which case
+        tha column of the table has no associated instance attribute).
+        The description must include 'ra', 'dec', and 'id' attributes
+        among the listed attributes.
+
+    data: `astropy.table.Table`, optional
+        A table containing all the instrumental aperture photometry results
+        for a given field on a given night.  If no data is passed,
+        an empty data table with the proper columns but no data is created.
+        If a table is provided, its format is checked against the
+        table_descript.
+
+        If data is passed in, it will be checked to make sure all columns
+        listed in table_descript exist, HOWEVER, additional columns in data
+        not listed in table_descript will NOT be removed.
+
+    Attributes
+    ----------
+    data: `astropy.table.Table`
+        A table formatted to match the table_descript formatting information
+        but containing no data.
+
+    dec: `astropy.table.Column`
+        A column of declination values with units of degrees
+
+    id: `astropy.table.Column`
+        A column of object id values (no specific format assumed)
+
+    ra: `astropy.table.Column`
+        A column of right ascension values with units of degrees
+
+    Other attributes may be created as defined by table_descript.
+    """
+
+    def __init__(self, table_descript=None, data=None):
```

Comment:
I don't think the `table_descript` is really optional.

```suggestion
    def __init__(self, table_descript, data=None):
```

Reply:
The None allowed easier catching of someone passing no inputs in for some reason.